pkg/autogen/v2/autogen.go

```diff
@@ -301,3 +301,30 @@ func (a *ImplAutogenV2) ExtractPodSpec(resource unstructured.Unstructured) (*uns
 
 	return &unstructured.Unstructured{Object: podSpec}, nil
 }
+
+type ExtractPodFunc func(resource *unstructured.Unstructured) (*unstructured.Unstructured, error)
+
+func getPodExtractor() ExtractPodFunc {
+	return func(resource *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+		return NewAutogenV2().ExtractPodSpec(*resource)
```

Comment:
I dont think you need to add an interface for `ExtractPodSpec`, `ComputeRules` is the only function that is used outside v2 package, ExtractPodSpec is just a part of the return values of `ComputeRules`, You don't need the ImplAutogenV2 struct, as there is no shared state those functions are using

You can just make `ExtractPodSpec` an independent function and just call that, also ExtractPodSpec might be better off as a private func for now